web/client/components/share/SharePanel.jsx

```diff
@@ -168,9 +168,9 @@ class SharePanel extends React.Component {
             shareUrl = `${shareUrl}${settings.markerEnabled ? "?marker=" : "?center="}${this.state.coordinate}&zoom=${this.state.zoom}`;
         }
         if (!settings.showSectionId && advancedSettings && advancedSettings.sectionId) {
-            const parsedUrl = shareUrl.split('/');
-            if (parsedUrl.length === 8) shareUrl = replace(shareUrl, parsedUrl[parsedUrl.length - 1], '');
-            if (parsedUrl.length === 9) shareUrl = replace(shareUrl, `${parsedUrl[parsedUrl.length - 2]}/${parsedUrl[parsedUrl.length - 1]}`, '');
+            const parsedUrl = shareUrl.split('#')[1]?.split('/');
+            if (parsedUrl.length === 5 && parsedUrl.includes('shared')) shareUrl = replace(shareUrl, parsedUrl[parsedUrl.length - 1], '');
+            if (parsedUrl.length === 6 && parsedUrl.includes('shared')) shareUrl = replace(shareUrl, `${parsedUrl[parsedUrl.length - 2]}/${parsedUrl[parsedUrl.length - 1]}`, '');
```

Comment:
The issue with the previus code was about the different path in the url between the local and the test enviroments. Now this changes ensure that the shareUrl is split on the hash before count the paresd parts of the path. This solutions seems to work but I would like to move this logic inside the [getSharedGeostoryUrl](https://github.com/geosolutions-it/MapStore2/blob/4ee11de7be140b78888fa304b6992a76e8feb5e2/web/client/utils/ShareUtils.js#L64-L68) function used some lines above https://github.com/geosolutions-it/MapStore2/blob/4ee11de7be140b78888fa304b6992a76e8feb5e2/web/client/components/share/SharePanel.jsx#L117

this is an example I did not test the code

in  ShareUtils.js
```js
getSharedGeostoryUrl: (url = '', removeSectionId) => {
    // check if the url is from geostory page
   // bacause share is used also in other page (eg: map viewer, dashboard)
    if (url.match(/\#\/(geostory)/)) {
        let geostoryUrl = url.match(/\/(geostory)\/((shared)|(newgeostory))/)
            ? url
            : url.replace('/geostory/', '/geostory/shared/');
        
        if (removeSectionId) {
            // implementation
        }
        return geostoryUrl;
    }

    return url;
}
```

in SharePanel.jsx

```js
const removeSectionId = !settings.showSectionId && advancedSettings && advancedSettings.sectionId;
let shareUrl = getSharedGeostoryUrl(removeQueryFromUrl(this.props.shareUrl), removeSectionId);

```

in this way we have all the logic for the geostory url in one function and we can add some tests.